pgctl/cli.py

```diff
@@ -410,8 +415,16 @@ def debug(self):
                 'Must debug exactly one service, not: ' + commafy(self.service_names),
             )
 
-        self.stop()
-        service.foreground()  # never returns
+        try:
+            self._entering_debug = True
+            if service.state['state'] != 'down':
+                self.stop()
+            else:
+                # If we were not previously running, then also run pre-start
+                self._run_playground_wide_hook('pre-start')
```

Comment:
This makes debug special with respect to pre-start. In the service already running case, the service will be stopped, and then started in foreground mode without pre-start running. It seems like pre-start should run in that case as well.